spire/journal/actions.py

```diff
@@ -1014,6 +1093,158 @@ async def update_journal_entry_tags(
+
+            values.append(insert_object)
+
+
+    # Deduplicate tags because of on_conflict_do_nothing not working with duplicates in values
+    seen = set()
+    deduplicated_values = []
+    for d in values:
+        t = tuple(sorted(d.items()))
+        if t not in seen:
+            seen.add(t)
+            deduplicated_values.append(d)
+
+    insert_statement = (
+        postgresql.insert(JournalEntryTag)
+        .values(deduplicated_values)
+        .on_conflict_do_nothing(index_elements=["journal_entry_id", "tag"])
+    )
+
+    try:
+        db_session.execute(insert_statement)
+        db_session.commit()
+    except:
+        db_session.rollback()
+        raise CommitFailed("Could not create tags")
+
+    return requested_entries
+
+
+async def delete_journal_entries_tags(
+    db_session: Session,
+    journal: Journal,
+    entries_tags_request: CreateEntriesTagsRequest,
+) -> List[UUID]:
+    
+
+    """
+    Delete tags for entries in journal.
+    """
+
+    requested_entries = [entry.journal_entry_id for entry in entries_tags_request.entries]
+
+    query = db_session.query(JournalEntry.id).filter(
+        JournalEntry.journal_id == journal.id
+    ).filter(JournalEntry.id.in_(requested_entries))
+
+
+    ### perfomance test https://stackoverflow.com/a/3462202/13271066
+
+    existing_entries: Set[UUID] = set([entry[0] for entry in query.all()])
+
+    diff = [x for x in requested_entries if x not in existing_entries]
+
+    if len(diff) > 0:
+        raise EntriesNotFound("Could not find some of the given entries" , diff)
+
+
+    values: List[Dict[str,Any]] = []
+
+    for entry_tag_request in entries_tags_request.entries:
+        entry_id = entry_tag_request.journal_entry_id
+
+        for tag in entry_tag_request.tags:
+
+            insert_object = {
+                "journal_entry_id": entry_id,
+                "tag": tag,
+            }
+
+            values.append(insert_object)
+
+
+    # Deduplicate tags
+
+    seen = set()
+    deduplicated_values = []
+    for d in values:
+        t = tuple(sorted(d.items()))
+        if t not in seen:
+            seen.add(t)
+            deduplicated_values.append(d)
+
+    selected_tags = (
+        db_session.query(
+            JournalEntryTag.id.label("id"),
+        )
+        .join(JournalEntry, JournalEntryTag.journal_entry_id == JournalEntry.id)
+        .filter(JournalEntry.journal_id == journal.id)
+        .filter(JournalEntryTag.journal_entry_id.in_([entry.journal_entry_id for entry in entries_tags_request.entries]))
+        .filter(JournalEntryTag.tag.in_([tag for entry in entries_tags_request.entries for tag in entry.tags]))
+        .cte("selected_tags")
+    )
+
+    delete_statement = (
+        db_session.query(JournalEntryTag)
+        .filter(JournalEntryTag.id.in_(select(selected_tags.c.id))
+        ).delete(synchronize_session=False)
+    )    
+
+    try:
+        db_session.commit()
+        logger.info(f"Deleted {delete_statement} tags")
```

Comment:
Add here pls journal id and num of entries like
`(f"Deleted {delete_statement} tags in journal {journal_id} for {len(entries)} entries")`